types/index.d.ts

```diff
@@ -0,0 +1,255 @@
+         *
+         * Usage: cachingStrategies.memoryHeavy
+         */
+        memoryHeavy: CachingStrategy;
+        /**
+         * Rankings will be cached by LRUCache. Requires optional dependency lru-cache.
+         *
+         * Usage: cachingStrategies.lru({...}, true/false)
+         *
+         * @param options This will be used as the options to create the LRUCache.
+         * @param compactCache Whether the cache should be optimized for lower RAM (true) or lower CPU (false).
+         */
+        lru: (options: any, compactCache: boolean) => CachingStrategy;
+    };
+    static rankingComparators: {
+        /**
+         * Rank everything by stat product descending then by attack descending.
+         * This is the default behavior, since in general, a higher stat product is usually preferable;
+         * and in case of tying stat products, higher attack means that you would be more likely to win CMP ties.
+         */
+        default: RankingComparator;
+        /**
+         * In addition to the default rules, also compare by CP descending in the end.
+         * While ties are not meaningfully different most of the time,
+         * the rationale here is that a higher CP looks more intimidating.
+         */
+        preferHigherCp: RankingComparator;
+        /**
+         * In addition to the default rules, also compare by CP ascending in the end.
+         * While ties are not meaningfully different most of the time,
+         * the rationale here is that you can flex beating your opponent using one with a lower CP.
+         */
+        preferLowerCp: RankingComparator;
+    };
+
+    /**
+     * Fetches the latest Pokemon data from Masterfile-Generator. Requires optional dependency node-fetch.
+     *
+     * @returns See pokemon-data.js for some helper methods.
+     * @see addPokemonDataHelpers
+     */
+    static fetchPokemonData(): Promise<PokemonData>;
+    /**
+     * Generates and fetches the latest Pokemon data from Pogo Data Generator. Requires optional dependency pogo-data-generator.
+     *
+     * @returns See pokemon-data.js for some helper methods.
+     * @see addPokemonDataHelpers
+     */
+    static fetchPokemonDataUnstable(): Promise<PokemonData>;
+
+    /**
+     * Filter the output of queryPvPRank with a subset of interested level caps.
+     *
+     * @param entries An array containing PvP combinations for a specific league from the output of queryPvPRank.
+     * @param interestedLevelCaps A non-empty array containing a list of interested level caps in ascending order.
+     * @returns The filtered array containing only capped entries and those whose cap matches the given caps.
+     */
+    static filterLevelCaps(entries: RankEntry[], interestedLevelCaps: number[]): RankEntry[];
+
+    /**
+     * Initialize your overlord Ohbem.
+     */
+    constructor(options?: OhbemOptions);
+
+    calculateAllRanks(stats: Stats, cpCap: number): Record<string, RankResult[][][]> | null;
```

Comment:
This return value is not correct, it could be two different values depending on if compact cache is enabled or not.